vhdbuilder/packer/install-dependencies.sh

```diff
@@ -351,28 +351,104 @@ INSTALLED_RUNC_VERSION=$(runc --version | head -n1 | sed 's/runc version //')
 echo "  - runc version ${INSTALLED_RUNC_VERSION}" >> ${VHD_LOGS_FILEPATH}
 capture_benchmark "${SCRIPT_NAME}_artifact_streaming_download"
 
-if [[ $OS == $UBUNTU_OS_NAME && $(isARM64) != 1 ]]; then  # no ARM64 SKU with GPU now
-  gpu_action="copy"
-  NVIDIA_DRIVER_IMAGE_SHA="20241008175307"
-  export NVIDIA_DRIVER_IMAGE_TAG="550.90.12-${NVIDIA_DRIVER_IMAGE_SHA}"
-  NVIDIA_DRIVER_IMAGE="mcr.microsoft.com/aks/aks-gpu-cuda"
-
-  mkdir -p /opt/{actions,gpu}
-  ctr -n k8s.io image pull $NVIDIA_DRIVER_IMAGE:$NVIDIA_DRIVER_IMAGE_TAG
-  if grep -q "fullgpu" <<< "$FEATURE_FLAGS"; then
-    bash -c "$CTR_GPU_INSTALL_CMD $NVIDIA_DRIVER_IMAGE:$NVIDIA_DRIVER_IMAGE_TAG gpuinstall /entrypoint.sh install"
-    ret=$?
-    if [[ "$ret" != "0" ]]; then
-      echo "Failed to install GPU driver, exiting..."
-      exit $ret
-    fi
+gpu_action=""
+declare -A pulled_gpu_images
+
+# Loop over each GPUContainerImage
+while IFS= read -r gpuImageToBePulled; do
+  # Extract 'cached' field and convert it to lowercase
+  cached=$(echo "${gpuImageToBePulled}" | jq -r '.cached' | tr '[:upper:]' '[:lower:]')
+
+  if [[ "$cached" != "true" ]]; then
+    # Skip images that are not meant to be cached
+    continue
   fi
 
-  cat << EOF >> ${VHD_LOGS_FILEPATH}
-  - nvidia-driver=${NVIDIA_DRIVER_IMAGE_TAG}
-EOF
+  # Extract 'osSelectors' if present
+  osSelectors=$(echo "${gpuImageToBePulled}" | jq -r '.osSelectors // empty')
+
+  shouldPull=0  # Default to not pull
+
+  if [[ -n "$osSelectors" ]]; then
+    # osSelectors is provided; check if current OS and arch match any entry
+    while IFS= read -r selector; do
+      os=$(echo "$selector" | jq -r '.os')
+      arch=$(echo "$selector" | jq -r '.arch')
+
+      if [[ "$os" == "$CURRENT_OS" ]]; then
+        if [[ "$arch" == "$CPU_ARCH" ]]; then
+          shouldPull=1
+          break  # Found a matching selector
+        fi
+      fi
+    done <<< "$(echo "$osSelectors" | jq -c '.[]')"
+  else
+    # No osSelectors provided; decide whether to pull
+    # Assuming we pull the image if no osSelectors are specified
+    shouldPull=1
+  fi
+
+  if [[ "$shouldPull" == "1" ]]; then
+    # Extract image details
+    downloadURL=$(echo "${gpuImageToBePulled}" | jq -r '.downloadURL')
+    imageName=$(echo "$downloadURL" | sed 's/:.*$//')
+
+    # Get the latestVersion
+    latestVersion=$(echo "${gpuImageToBePulled}" | jq -r '.multiArchVersionsV2[0].latestVersion')
```

Comment:
If you only put [0], later on when you add another multiArchVersionsV2[1] block in components.json, it won't capture it.
Suggested to loop the whole `multiArchVersionsV2` array. And if you only have 1 multiArchVersionsV2 entry in components.json, it will just work like using [0].

Reply:
1 example for your reference: 
At around ln414.
```  
  MULTI_ARCH_VERSIONS=()
  updateMultiArchVersions "${imageToBePulled}"
```

Reply:
Thanks, updated it to reuse that function

Reply:
Actually, under no circumstances would we need to pull two CUDA images for the same VHD

Reply:
If that's the case, maybe create a `gpuImageVersion` and it's just simple struct instead of an array of struct? Then it's much cleaner and explicit enough. Otherwise one day someone will ask, why we were only picking the first element of an array, but we didn't just create a non-array struct? Does that sound making sense to you?

Reply:
Planning to change it to so we constrain it to one gpuVersion per GPUContainerImage: 

#GPUContainerImage: {
	downloadURL: string
	gpuIVersion:   #VersionV2
	cached: bool
	osSelectors?:         [...#OSSelector]
}